bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/EventFilterHandler.java

```diff
@@ -0,0 +1,362 @@
+import org.openhab.binding.icalendar.internal.handler.PullJob.CalendarUpdateListener;
+import org.openhab.binding.icalendar.internal.logic.AbstractPresentableCalendar;
+import org.openhab.binding.icalendar.internal.logic.Event;
+import org.openhab.binding.icalendar.internal.logic.EventTextFilter;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelGroupUID;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EventFilterHandler} filters events from a calendar and presents them in a dynamic way.
+ *
+ * @author Michael Wodniok - Initial Contribution
+ */
+@NonNullByDefault
+public class EventFilterHandler extends BaseThingHandler implements CalendarUpdateListener {
+
+    private @Nullable EventFilterConfiguration configuration;
+    private final Logger logger = LoggerFactory.getLogger(EventFilterHandler.class);
+    private final List<ResultChannelSet> resultChannels;
+    private @Nullable ScheduledFuture<?> updateFuture;
+    private boolean initFinished;
+
+    public EventFilterHandler(Thing thing) {
+        super(thing);
+        resultChannels = new CopyOnWriteArrayList<>();
+        initFinished = false;
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            updateStates();
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        final ScheduledFuture<?> currentUpdateFuture = updateFuture;
+        if (currentUpdateFuture != null) {
+            currentUpdateFuture.cancel(true);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            if (initFinished) {
+                updateStates();
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        @Nullable
```

Comment:
The compiler is more intelligent on local variables. This annotation could be removed.

Reply:
OK, doing locally